### ISSUE TITLE
warning that duecredit is not installed when loading package

### ISSUE THREAD

**Issue by kain88-de** [MDAnalysis/mdanalysis#1872]
### Expected behaviour

no warning when loading the module

### Actual behaviour

warning that duecredit is not installed.

### Currently version of MDAnalysis:
(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
0.18.0

**Comment by orbeckst** [comment]
I thought we had taken care of this as part of PR #1822 , specifically https://github.com/MDAnalysis/mdanalysis/pull/1822/commits/e3d5cace73f3e0cec1b14792291886673938cfab but it seems we have to revisit.

**Comment by orbeckst** [comment]
For the record (using Python 3.6 on macOS) in a conda environment *without duecredit* gives:

```bash
cat > credit.py << 'EOF'
import MDAnalysis
print(MDAnalysis.__version__)
EOF
python credit.py
```

gives
```
/[...]/mdanalysis/package/MDAnalysis/due.py:88: UserWarning: No module named 'duecredit'
  warnings.warn(str(err))
0.18.1-dev
```

**Comment by orbeckst** [comment]
It raises a `ModuleNotFoundError: No module named 'duecredit'` and because
```python
>>> isinstance(ModuleNotFoundError, ImportError)
False
```
it is not caught. (I could have sworn that `ModuleNotFoundError` is an instance of `ImportError`...).

Will patch it. Not sure how to test it in our current testing framework because the `due.py` code is run on the very first import of MDAnalysis. Maybe mock `duecredit` and re-import `MDAnalysis.due`... suggestions welcome.

**Comment by richardjgowers** [comment]
@orbeckst we used to have some tests that tested for deprecation warnings or something, I think we deleted the dict entry from `sys.modules` and that forced the reload of the module

**Comment by orbeckst** [comment]
Btw, the Python docs say https://docs.python.org/3/library/exceptions.html#ModuleNotFoundError

>  exception ModuleNotFoundError
>
>    A subclass of ImportError which is raised by import when a module could not be located. It is also raised when None is found in sys.modules.

The following does catch that `ModuleNotFoundError`
```python
In [12]: try:
    ...:     raise ModuleNotFoundError("foo")
    ...: except ImportError as err:
    ...:     print("Import error", err)
    ...:
Import error foo
```

**Comment by orbeckst** [comment]
I just can't read code anymore... the exceptions work as they should, I just have to remove the debugging code :-p https://github.com/MDAnalysis/mdanalysis/blob/8c331d7883ccf4a61dec884f2d95110726ce7573/package/MDAnalysis/due.py#L88

**Comment by mimischi** [comment]
I'm still getting the same warning with MDAnalysis 0.18.0 in a fresh environment:
```bash
$ python -c "import MDAnalysis as mda; import sys; print(mda.__version__); print(sys.version);"
/home/mischi/.virtualenvs/test-oN5qujzB/lib/python3.6/site-packages/MDAnalysis/due.py:88: UserWarning: No module named 'duecredit'
  warnings.warn(str(err))
0.18.0
3.6.5 (default, Apr 14 2018, 13:17:30) 
[GCC 7.3.1 20180406]
```

**Comment by kain88-de** [comment]
This error is fixed in the next release

**Comment by rainbeeliu** [comment]
I first saw the same warning. But after I install the `duecredit` package, the warning message disappears. I guess the package dependencies are not right?

**Comment by richardjgowers** [comment]
@rainbeeliu the dependencies are correct in that `duecredit` isn't necessary for the package to function.  You only need duecredit for duecredit (citation) things.
